src/main/java/polyglot/ServiceList.java

```diff
@@ -0,0 +1,108 @@
+package polyglot;
+
+import java.io.File;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import com.google.common.base.Joiner;
+import com.google.protobuf.DescriptorProtos.FileDescriptorSet;
+import com.google.protobuf.Descriptors.Descriptor;
+import com.google.protobuf.Descriptors.FieldDescriptor;
+import com.google.protobuf.Descriptors.MethodDescriptor;
+import com.google.protobuf.Descriptors.ServiceDescriptor;
+
+import polyglot.protobuf.ServiceResolver;
+
+/** Simple utility to list the services, methods and message definitions for the known GRPC end-points */
+public class ServiceList {
+
+  /** Lists the set of GRPC services - optionally filtered by service name (contains) or method name (contains) */ 
+  public static void listServices(FileDescriptorSet fileDescriptorSet, String protoDiscoveryRoot,
+      Optional<String> serviceFilter, Optional<String> methodFilter, Optional<Boolean> withMessage) {
+
+    ServiceResolver serviceResolver = ServiceResolver.fromFileDescriptorSet(fileDescriptorSet);
+
+    // Add white-space before the rendered output
+    System.out.println();
```

Comment:
So far, we have been using the logger for all output. That approach needs some re-thinking, but we should probably avoid "System.out.println" for now for consistency (unless you think this is unreasonable for your use case). Thoughts?